packages/button/theme/lumo/vaadin-button-styles.js

```diff
@@ -75,14 +75,9 @@ const button = css`
 
   /* Hover */
 
-  :host(:hover)::before {
-    opacity: 0.05;
-  }
-
-  /* Disable hover for touch devices */
-  @media (pointer: coarse) {
-    :host(:not([active]):hover)::before {
-      opacity: 0;
+  @media (any-hover: hover) {
+    :host(:hover)::before {
+      opacity: 0.02;
```

Comment:
The previous implementation was done reverse because Firefox didn't support the hover or pointer media queries back in the day. Now all the supported browsers can handle the hover media query.